File: leetcode/easy/203_移除链表元素/203_移除链表元素.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        dummy_head = ListNode(0)  # 设置一个虚拟头结点
        dummy_head.next = head    # 将虚拟头结点指向head，这样方便后面做删除操作
        cur = dummy_head
        while cur.next is not None:
            if cur.next.val == val:
                cur.next = cur.next.next  # 删除cur.next节点
            else:
                cur = cur.next
        return dummy_head.next

File: leetcode/easy/203_移除链表元素/203_移除链表元素.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
 func removeElements(head *ListNode, val int) *ListNode {
    dummyHead := &ListNode{Next: head}
    cur := dummyHead
    for cur.Next != nil {
        if cur.Next.Val == val {
            cur.Next = cur.Next.Next  // 删除cur.Next节点
        } else {
            cur = cur.Next
        }
    }
    return dummyHead.Next
}